### PR TITLE
Adding support for APL render document and user event

### PR DESCRIPTION
**Pull request by brightsparc**
* **Please check if the PR fulfills these requirements**

- [x] The commit message describes your change
- [x] Tests for the changes have been added if possible (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Changes are mentioned in the changelog (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add support for APL render document validation, and UserEvent touch events.

* **What is the current behavior?** (You can also link to an open issue here)
This is a new features.

* **What is the new behavior (if this is a feature change)?**
Ability to support APL templates and user events.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)
No breaking changes.

* **Other information**:

### CODE REVIEW

File: lib/tester/AplValidator.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export class AplValidator extends ResponseValidator {
+    public validate(currentItem : SequenceItem, response : ResponseEnvelope) : void {
+        if (currentItem.renderDocument) {
+            if (!response.response.directives) {
+                fail('the response did not contain any directives');
+            }
+            const renderedDocument = <interfaces.alexa.presentation.apl.RenderDocumentDirective> response.response.directives.find((value) => value.type === 'Alexa.Presentation.APL.RenderDocument');
+            if (!renderedDocument) {
+                fail('the response did not play a stream');
```
Comment (hoegertn) [inline lib/tester/AplValidator.ts]: Wrong error message

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 
 ## current master
+* feat: add `renderDocument` parameter used to asset templates in `AplValidator` class.
```
Comment (hoegertn) [inline CHANGELOG.md]: assert

**Review by hoegertn — CHANGES_REQUESTED** [review]
Thanks. This looks great. Please fix the two minor typos. I will merge it then.

**Review by hoegertn — APPROVED** [review]

**Comment by brightsparc** [comment]
Hi @hoegertn are you able to push this latest update to NPM repo?

Also I notice the docs are slight out of date if you wanted to do `npm run build` as part of that push.

**Comment by hoegertn** [comment]
Release as 2.5.0

The `npm run build` is done automatically on release.

Thanks for reminding me.

**Pull request merged**